components/org.wso2.carbon.identity.client.attestation.filter/src/main/java/org/wso2/carbon/identity/client/attestation/filter/ClientAttestationProxy.java

```diff
@@ -0,0 +1,282 @@
+            // Retrieve the 'response_mode' parameter from the request.
+            String responseMode = bodyContentParams.get(RESPONSE_MODE).get(0).toString();
+            return responseMode.equalsIgnoreCase(DIRECT);
+        }
+        return false;
+    }
+
+    /**
+     * Retrieve body content as a String, List map.
+     *
+     * @param message JAX-RS incoming message
+     * @return Body parameter of the incoming request message
+     */
+    protected Map<String, List> getContentParams(Message message) {
+
+        Map<String, List> contentMap = new HashMap<>();
+        List contentList = message.getContent(List.class);
+        contentList.forEach(item -> {
+            if (item instanceof MetadataMap) {
+                MetadataMap metadataMap = (MetadataMap) item;
+                metadataMap.forEach((key, value) -> {
+                    if (key instanceof String && value instanceof List) {
+                        contentMap.put((String) key, (List) value);
+                    }
+                });
+            }
+        });
+        return contentMap;
+    }
+
+    private String getTenantDomain() {
+
+        return Optional.ofNullable(IdentityTenantUtil.getTenantDomainFromContext())
+                .filter(StringUtils::isNotBlank)
+                .orElseGet(() -> PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantDomain());
+    }
+
+    /**
+     * Sets the Client Attestation context to the HttpServletRequest's attributes.
+     *
+     * @param request                   The HttpServletRequest to which the context should be added.
+     * @param clientAttestationContext  The Client Attestation context to be added to the request.
+     */
+    private void setContextToRequest(HttpServletRequest request, ClientAttestationContext clientAttestationContext) {
+        // Check if DEBUG logging is enabled before logging
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Setting Client Attestation context to request");
+        }
+        // Add the Client Attestation context as an attribute to the HttpServletRequest
+        request.setAttribute(CLIENT_ATTESTATION_CONTEXT, clientAttestationContext);
+    }
+
+    /**
+     * Retrieves the service provider based on the given client ID and tenant domain.
+     *
+     * @param clientId     The client ID associated with the service provider.
+     * @param tenantDomain The tenant domain in which the service provider is registered.
+     * @return The retrieved service provider.
+     * @throws WebApplicationException If an error occurs during the retrieval process.
+     */
+    private ServiceProvider getServiceProvider(String clientId, String tenantDomain) {
+
+        ServiceProvider serviceProvider;
+        try {
+            serviceProvider = applicationManagementService.getServiceProviderByClientId(clientId, OAUTH2, tenantDomain);
+        } catch (IdentityApplicationManagementException e) {
```

Comment:
At which point can this error occur? if it occurs for invalid client id lets log the exception as a debug log. if it can occur only for server exceptions add a error log with the exception. same should be considered when sending the error response.

Reply:
As discussed, we are going to throw the exception from proxy.

Reply:
Lets add a catch clause for IdentityApplicationManagementClientException